update scummvm core info

spleen1981 opened the pull request.

hizzlekizzle commented:
look at all those file extensions simplified down to 1...

Can we sneak an updated description field in there while we're at it? Maybe something like:

> description = "The ScummVM adventure game engine ported to libretro. This core is built directly from the upstream repo and is synced upon stable releases, though it is not supported upstream, so if you run into problems, please report them to libretro and/or make sure they occur in the standalone ScummVM program, as well, before making any reports to them.

RobLoach commented on scummvm_libretro.info:
```diff
@@ -1,15 +1,15 @@
 display_name = "ScummVM"
 authors = "SCUMMVMdev"
-supported_extensions = "0|1|2|3|4|5|6|8|25|99|101|102|455|512|scummvm|scumm|gam|z5|dat|blb|z6|ROM|001|taf|zblorb|dcp|(a)|cup|HE0|(A)|D$$|STK|z8|hex|ITK|CD1|pic|Z5|z3|blorb|ulx|DAT|cas|CGA|PIC|acd|SYS|OVL|alr|t3|gblorb|tab|AP|CRC|EXE|z4|W32|MAC|mac|WIN|003|000|exe|asl|slg|AVD|INI|SND|cat|ANG|CUP|SYS16|img|LB|TLK|MIX|RLB|#02|FNT|win|HE1|DMU|FON|SCR|MAP|TEX|HEP|DIR|DRV|a3c|GRV|CUR|CC|COD|OPT|LA0|gfx ...
+supported_extensions = "scummvm"
```
Note that the [ScummVM.dat database](https://github.com/libretro/libretro-database/blob/master/dat/ScummVM.dat) does allow scanning and loading ScummVM content directly from playlists.

spleen1981 replied:
For info, the core info file can now be obtained from the libretro core with `make coreinfo`. Extensions are retrieved from latest ScummVM.dat, "firmwares" from ScummVM actual source.

RobLoach replied:
That's really cool! :+1:  Thanks so much.

spleen1981 commented:
> Can we sneak an updated description field in there while we're at it? Maybe something like:
> 
> > description = "The ScummVM adventure game engine ported to libretro. This core is built directly from the upstream repo and is synced upon stable releases, though it is not supported upstream, so if you run into problems, please report them to libretro and/or make sure they occur in the standalone ScummVM program, as well, before making any reports to them.

sounds good, added and squashed
